src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -798,31 +853,206 @@ let replaceOutputColumn (annotationTableName:string) (existingOutputColumn: Buil
+            else                
+                $"Found existing output column \"{oldOutputColumn}\". Changed output column to \"{annotationTable.columns.items.[(int) outputColumn.Value.index].name}\"."
+
+        let msg = InteropLogging.Msg.create InteropLogging.Warning warningMsg
+
+        let loggingList = [
+            msg
+        ]
+
+        loggingList
+    else
+        failwith "Something went wrong! The update output column is not filled with data! Please report this as a bug to the developers."
+
+/// <summary>
+/// Add a new outputcolumn to an annotation table
+/// </summary>
+/// <param name="annotationTable"></param>
+/// <param name="arcTable"></param>
+/// <param name="newBB"></param>
+let addOutputColumn (annotationTable:Table) (arcTable:ArcTable) (newBB:CompositeColumn) =
+
+    if arcTable.TryGetOutputColumn().IsSome then
+        failwith "Something went wrong! The add output column is filled with data! Please report this as a bug to the developers."
+
+    else
+
+        let rowCount = arcTable.RowCount + 1
+
+        let createCol name =
+            let col value = createMatrixForTables 1 rowCount value
+            annotationTable.columns.add(
+                index   = -1.,
+                values  = U4.Case1 (col ""),
+                name    = name
+            )
+
+        let newColumn = createCol (newBB.Header.ToString())
+        let columnBody = newColumn.getDataBodyRange()
+        // Fit column width to content
+        columnBody.format.autofitColumns()
+
+        let msg = InteropLogging.Msg.create InteropLogging.Info $"Added new output column: {newBB.Header}"
+
+        let loggingList = [
+                msg
+        ]
+
+        loggingList
+
+/// Handle any diverging functionality here. This function is also used to make sure any new building blocks comply to the swate annotation-table definition.
+let addAnnotationBlockHandler (newBB:CompositeColumn) =
+    Excel.run(fun context ->
+        promise {
+
+            //let activeSheet = context.workbook.worksheets.getActiveWorksheet()
+            //Try to get the name of the currently active sheet
+            let! annotationTableName = tryGetActiveAnnotationTableName context
+
+            //When a name is available get the annotation and arctable for easy access of indices and value adaption
+            //Annotation table enables a easy way to adapt the table, updating existing and adding new columns
+            let! annotationTable =
+                if annotationTableName.IsSome then tryGetAnnotationTableByName context annotationTableName.Value
+                else tryGetAnnotationTableByName context ""
+
+            //Arctable enables a fast check for the existence of input- and output-columns and their indices
+            let! arcTable =
```

Comment:
This should be based on annotationTable and not request from context again.
Then use Option.map for parsing and only ever check if arcTable.IsSome

Reply:
fixed